### ISSUE TITLE
no appropriate error when AuthenticationFailed due to consistency level does not match(system_auth)

### ISSUE THREAD

**Issue by abvgedeika** [scylladb/scylladb#2339]
*Installation details*
Scylla version (or git commit hash): 666.development-0.20170427.14b9aa2
Cluster size: 2
OS (RHEL/CentOS/Ubuntu/AWS AMI): ubuntu16

1) start cluster with 2 nodes with config
config = {'authenticator': 'org.apache.cassandra.auth.PasswordAuthenticator',
'authorizer': 'org.apache.cassandra.auth.CassandraAuthorizer',
'permissions_validity_in_ms': permissions_validity}

2) set RF=2:
alter keyspace system_auth with replication = {'class': 'org.apache.cassandra.locator.SimpleStrategy', 'replication_factor':2};

3) stop/remove one node from cluster, try to connect to  another one

scylla doesn't contain any info that AuthenticationFailed due to consistency level does not match
:
```
 cqlsh 127.0.0.1 -u cassandra -p cassandra
Connection error: ('Unable to connect to any servers', {'127.0.0.1': AuthenticationFailed('Failed to authenticate to 127.0.0.1: Error from server: code=0100 [Bad credentials] message="authentication failed"',)})
```

c*
```
 ./cassandra/bin/cqlsh 127.0.0.1 -u cassandra -p cassandra
Connection error: ('Unable to connect to any servers', {'127.0.0.1': AuthenticationFailed('Failed to authenticate to 127.0.0.1: Error from server: code=0100 [Bad credentials] message="org.apache.cassandra.exceptions.UnavailableException: Cannot achieve consistency level QUORUM"',)})
```

**Comment by mykaul** [comment]
@ptrsmrn - to be closed when we move system_auth to Everywhere / Raft?

**Comment by ptrsmrn** [comment]
@KrzaQ could you please reproduce this scenario to see if we are still having a mismatch between Scylla vs Cassandra regarding what error message is being returned to the user?

**Comment by ptrsmrn** [comment]
> @ptrsmrn - to be closed when we move system_auth to Everywhere / Raft?

@mykaul I am not sure (and also I'm not that familiar with auth yet, so take my reply with a pinch of salt). The auth sources mention either CL=ONE or CL=QUORUM. Having 1 out of 2 nodes alive doesn't meet quorum requirements, and it seems like we at least [require CL=QUORUM for admin users](https://github.com/scylladb/scylladb/blob/master/auth/standard_role_manager.cc#L83), so that wouldn't work for Everywhere strategy (?). OTOH with RAFT, which seems to be more desired for auth (failed to find the ticket to support that), there should be no need to CL=QUORUM, because all the nodes would have (roughly speaking) the same data.

**Comment by ptrsmrn** [comment]
> @ptrsmrn - to be closed when we move system_auth to Everywhere / Raft?

Yes, per Eliran - https://github.com/scylladb/scylladb/issues/15251#issuecomment-1704060211

**Comment by KrzaQ** [comment]
Tested with the current versions

Cassandra [latest](https://hub.docker.com/layers/library/cassandra/latest/images/sha256-3146036995c4466a848810ad8a368bdae48cca88edcc4b39debea9f2e45c053f?context=explore):
> Connection error: ('Unable to connect to any servers', {'127.0.0.1:9042': AuthenticationFailed('Failed to authenticate to 127.0.0.1:9042: Error from server: code=0100 [Bad credentials] message="Unable to perform authentication: Cannot achieve consistency level QUORUM"')})

Scylla 5.2:
> Connection error: ('Unable to connect to any servers', {'172.21.0.2:9042': AuthenticationFailed('Failed to authenticate to 172.21.0.2:9042: Error from server: code=0100 [Bad credentials] message="authentication failed"',)})

**Comment by ptrsmrn** [comment]
Thanks Paweł. I think w can fix it. Depending on what message scylla's auth module is returning to cqlsh, either we have to change scylla code or cqlsh code.

**Comment by kostja** [comment]
@KrzaQ I think "topology_password_auth" is not the best name, "topology" stands for the topology tests, i.e. tests which cover topology changes such as adding or removing nodes. You could call your suite "auth"  or "auth_cluster" or "auth_many", something along the lines, and @ptrsmrn team will be owning the suite going forward.

**Comment by mykaul** [comment]
@KrzaQ - I think there's a breakage in the tests now - please see https://jenkins.scylladb.com/job/scylla-5.4/job/dtest-release/8/testReport/auth_test/TestAuth/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split004___test_remove_dead_node_consistency_failed/

**Comment by mykaul** [comment]
> @KrzaQ - I think there's a breakage in the tests now - please see https://jenkins.scylladb.com/job/scylla-5.4/job/dtest-release/8/testReport/auth_test/TestAuth/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split004___test_remove_dead_node_consistency_failed/

@KrzaQ , @ptrsmrn - please see if https://github.com/scylladb/scylla-dtest/pull/3741 fixes it. If so, let's approve it and get it backported ASAP. Or revert the offending PR.

**Comment by bhalevy** [comment]
> > @KrzaQ - I think there's a breakage in the tests now - please see https://jenkins.scylladb.com/job/scylla-5.4/job/dtest-release/8/testReport/auth_test/TestAuth/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split004___test_remove_dead_node_consistency_failed/
> 
> @KrzaQ , @ptrsmrn - please see if [scylladb/scylla-dtest#3741](https://github.com/scylladb/scylla-dtest/pull/3741) fixes it. If so, let's approve it and get it backported ASAP. Or revert the offending PR.

The dtest fix assumes this issue is fixed, but it is not backported anywhere yet.

The test was marked with `require('2339')`, and while this issue was fixed only on master, it started running in all branches since the DTEST_REQUIRE logic doesn't know which branch is fixed and which is not.

We can change the test in the old branches to expect the unfixed behavior (with the less-informative error message) or skip it until the fix for this issue is backported.
I think that the former approach would be better since the test will start failing once the backport is done, and then it can be fixed to match the new expected behavior.

Cc @roydahan, @fruch
This issue was fixed only in master, and the test started running everywhere.

**Comment by roydahan** [comment]
Yes.
In the past the check for required vs closed was implemented in the pipeline so it will affect only master branch.
But since then a lot has been changed and this code reached to all branches.

I suggest that the fix in https://github.com/scylladb/scylla-dtest/issues/3733 will take care of it (do your first suggestion and will verify the "bad" error message based on the version till the fix will land in these old branches)

**Comment by bhalevy** [comment]
> Yes. In the past the check for required vs closed was implemented in the pipeline so it will affect only master branch. But since then a lot has been changed and this code reached to all branches.
> 
> I suggest that the fix in [scylladb/scylla-dtest#3733](https://github.com/scylladb/scylla-dtest/issues/3733) will take care of it (do your first suggestion and will verify the "bad" error message based on the version till the fix will land in these old branches)

We can still control it in the pipeline.
`fixture_require_version` treats the `DTEST_REQUIRE` environment variable as tri-state:
- `disabled` - tests always run
- `auto` (the default) - tests run only if issue is closed on github
- `enabled` - tests are always skipped

We can set `DTEST_REQUIRE=enabled` in the pipeline in the branches to skip all cases marked with `pytest.mark.require`.
Then we should remove old markers in branches to let the tests run again.

I think that the procedure for forking a branch in dtest should include scrubbing the require markers
and removing all that are already fixed at the time of the branch since the branch should refer to a frozen state of the respective scylla branch.
Later on, when fixes are backported to scylla branches, we should remove the respective require marker from the respective dtest branches.

**Comment by avikivity** [comment]
Backported to 5.2, 5.4.
